mbed_lstools/platform_database.py

```diff
@@ -0,0 +1,341 @@
+    u'9008': u'XADOW_M0',
+    u'9009': u'ARCH_BLE',
+    u'9010': u'ARCH_GPRS',
+    u'9011': u'ARCH_MAX',
+    u'9012': u'SEEED_TINY_BLE',
+    u'9900': u'NRF51_MICROBIT',
+    u'C002': u'VK_RZ_A1H',
+    u'FFFF': u'K20 BOOTLOADER',
+    u'RIOT': u'RIOT',
+}
+
+class PlatformDatabase(object):
+    """Represents a union of multiple platform database files.
+    Handles inter-process synchronization of database files.
+    """
+
+    target_id_pattern = re.compile(r'^[a-fA-F0-9]{4}$')
+
+    def __init__(self, database_files, primary_database=None):
+        """Construct a PlatformDatabase object from a series of platform database files"""
+        self._prim_db = primary_database
+        if not self._prim_db and len(database_files) == 1:
+            self._prim_db = database_files[0]
+        self._dbs = OrderedDict()
+        self._keys = set()
+        for db in database_files:
+            try:
+                new_db = json.load(open(db, encoding="utf-8"))
+                duplicates = set(self._keys).intersection(set(new_db.keys()))
+                if duplicates:
+                    logger.warning(
+                        "Duplicate platform ids found: %s,"
+                        " ignoring the definitions from %s",
+                        " ".join(duplicates), db)
+                self._dbs[db] = new_db
+                self._keys = self._keys.union(new_db.keys())
+            except IOError as exc:
+                if db is LOCAL_PLATFORM_DATABASE:
+                    logger.warning(
+                        "Could not open local platform database at %s; "
+                        "Recereating", db)
+                    try:
+                        makedirs(dirname(db))
+                    except OSError:
+                        pass
+                    with open(db, "w", encoding="utf-8") as out:
+                        out.write(unicode(json.dumps(DEFAULT_PLATFORM_DB)))
+                    new_db = copy(DEFAULT_PLATFORM_DB)
+                    self._dbs[db] = new_db
+                    self._keys = self._keys.union(new_db.keys())
+                else:
+                    if db is not self._prim_db:
+                        logger.error(
+                            "Could not open platform database %s: %s; skipping",
+                            db, str(exc))
+                    self._dbs[db] = {}
+            except ValueError as exc:
+                logger.error(
+                    "Could not parse platform database %s because %s; "
+                    "skipping", db, str(exc))
+                self._dbs[db] = {}
+
+    def items(self):
+        for db in self._dbs.values():
+            for entry in db.items():
+                yield entry
+
+    def all_ids(self):
+        return iter(self._keys)
+
+    def get(self, index, default=None):
+        """Standard lookup function. Works exactly like a dict"""
+        for db in self._dbs.values():
+            maybe_answer = db.get(index, None)
```

Comment:
No critique, just the variable name made me happy 😄

Reply:
Yeah, it might not be an answer, so maybe!